### ISSUE TITLE
result of get_db_path should be a part of ClientConfig

### ISSUE THREAD

**Issue by debris** [ethcore/parity#1137]
I'm opening it up for discussion. Currently `get_db_path` is used in config, client

```
ethcore/src/client/client.rs:111:pub fn get_db_path(path: &Path, pruning: journaldb::Algorithm, genesis_hash: H256) -> PathBuf {
ethcore/src/client/client.rs:130:		let path = get_db_path(path, config.pruning, spec.genesis_header().hash());
parity/configuration.rs:29:use ethcore::client::{append_path, get_db_path, ClientConfig, Switch, VMType};
parity/configuration.rs:188:			let db = journaldb::new(&append_path(&get_db_path(Path::new(&self.path()), *i, spec.genesis_header().hash()), "state"), *i);
```

and another one time in migration code in this pr https://github.com/ethcore/parity/pull/1014 

We could simplify it by making `db_path` part of the config.

**Comment by gavofyork** [comment]
seems reasonable.

**Comment by debris** [comment]
fixed in #1600

**Comment by rphmeier** [comment]
in the #1679, i alter `get_db_path` to accept the path with the genesis hash already appended, so it really just figures out what the path should be based on pruning
